sklearn/datasets/tests/test_svmlight_format.py

```diff
@@ -153,13 +150,13 @@ def test_load_invalid_order_file():
 
 
 def test_load_zero_based():
-    f = BytesIO(b("-1 4:1.\n1 0:1\n"))
+    f = BytesIO("-1 4:1.\n1 0:1\n".encode())
     assert_raises(ValueError, load_svmlight_file, f, zero_based=False)
 
 
 def test_load_zero_based_auto():
-    data1 = b("-1 1:1 2:2 3:3\n")
-    data2 = b("-1 0:0 1:1\n")
+    data1 = "-1 1:1 2:2 3:3\n".encode()
+    data2 = "-1 0:0 1:1\n".encode()
```

Comment:
Probably better to replace all the .encode by `b"<some string>"` here and below in this file.